DirectProgramming/DPC++FPGA/Tutorials/Features/prototype/CSR_and_streaming_interfaces/README.md

```diff
@@ -0,0 +1,300 @@
+    ...
+  }
+};
+```
+
+### Default Interfaces
+If no annotation is specified for the kernel control, then a CSR agent kernel ontrol interface will be inferred by the compiler. If no annotation is specified for the a kernel argument, then that kernel argument will have the same interface as the kernel control interface. In the Lambda programming model, all kernel arguments will have the same interface as the kernel control interface.
+
+### Testing the Tutorial
+In `CSR_and_streaming_interfaces.cpp`, two kernels are decalred using the functor programming model, one with CSR Agent kernel control interface, and one with streaming kernel control interface.
+
+```c++
+struct CSRAgentControlIP {
+  register_map ValueT *input;
+  ValueT *output;
+  conduit size_t n;
+  CSRAgentControlIP(ValueT *in_, ValueT *out_, size_t N_)
+      : input(in_), output(out_), n(N_) {}
+  register_map_interface void operator()() const {
+    for (int i = 0; i < n; i++) {
+      output[i] = SomethingComplicated(input[i]);
+    }
+  }
+};
+struct StreamingControlIP {
+  conduit ValueT *input;
+  conduit ValueT *output;
+  size_t n;
+  StreamingControlIP(ValueT *in_, ValueT *out_, size_t N_)
+      : input(in_), output(out_), n(N_) {}
+  streaming_interface void operator()() const {
+    for (int i = 0; i < n; i++) {
+      output[i] = SomethingComplicated(input[i]);
+    }
+  }
+};
+```
+
+These two kernels are invoked in the same way in the host code, by submitting a single_task into the sycl queue.
+
+```c++
+template <typename KernelType>
+void KernelTest(sycl::queue& q, ValueT* in, ValueT* out, size_t count) {
+  q.single_task(KernelType{in, out, count}).wait();
+}
+```
+
+## Key Concepts
+* Basics of declaring kernel control interfaces and kernel argument interfaces
+
+## License
+Code samples are licensed under the MIT license. See
+[License.txt](https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt) for details.
+
+Third party program Licenses can be found here: [third-party-programs.txt](https://github.com/oneapi-src/oneAPI-samples/blob/master/third-party-programs.txt)
+
+## Building the `CSR_and_streaming_interfaces` Tutorial
+
+> **Note**: If you have not already done so, set up your CLI
+> environment by sourcing  the `setvars` script located in
+> the root of your oneAPI installation.
+>
+> Linux Sudo: `. /opt/intel/oneapi/setvars.sh`
+>
+> Linux User: `. ~/intel/oneapi/setvars.sh`
+>
+> Windows: `C:\Program Files(x86)\Intel\oneAPI\setvars.bat`
+>
+>For more information on environment variables, see Use the setvars Script for [Linux or macOS](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html), or [Windows](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-windows.html).
+
+
+### Include Files
```

Comment:
Please remove this Include files section entirely

Reply:
removed